.github/scripts/shellCheck.sh

```diff
@@ -1,48 +1,39 @@
 #!/bin/bash
 
-CURRENT_DIR=$(pwd)
-ROOT_DIR="$(dirname "$(dirname "$(dirname "$(realpath "${BASH_SOURCE[0]}")")")")"
+ROOT_DIR=$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")/../.." &>/dev/null && pwd)
+readonly ROOT_DIR
 
-cd "$ROOT_DIR" || exit 1
+source "$ROOT_DIR"/.github/scripts/shellUtils.sh
 
-source ./.github/scripts/shellUtils.sh
-
-declare -r DIRECTORIES_TO_IGNORE=(
-  './node_modules'
-  './vendor'
-  './ios/Pods'
-  './.husky'
-)
+readonly DIRECTORIES_TO_IGNORE="\
+-path $ROOT_DIR/node_modules \
+-o -path $ROOT_DIR/vendor \
+-o -path $ROOT_DIR/ios/Pods \
+-o -path $ROOT_DIR/.husky"
 
 # This lists all shell scripts in this repo except those in directories we want to ignore
-read -ra IGNORE_DIRS < <(join_by_string ' -o -path ' "${DIRECTORIES_TO_IGNORE[@]}")
-SHELL_SCRIPTS=$(find . -type d \( -path "${IGNORE_DIRS[@]}" \) -prune -o -name '*.sh' -print)
-info "👀 Linting the following shell scripts using ShellCheck: $SHELL_SCRIPTS"
-info
+# shellcheck disable=SC2086
+SHELL_SCRIPTS="$(find "$ROOT_DIR" -type d \( $DIRECTORIES_TO_IGNORE \) -prune -o -name '*.sh' -print)"
```

Comment:
You don't need the `-print` flag, it's the default when you're not providing something like `-exec` that overrides it

```suggestion
SHELL_SCRIPTS="$(find "$ROOT_DIR" -type d \( $DIRECTORIES_TO_IGNORE \) -prune -o -name '*.sh')"
```

Reply:
My testing shows that this is necessary. Perhaps the `-prune` is changing the default behavior? If I remove it, it does not ignore `node_modules` as expected.

Reply:
Hm interesting, you're right. If you remove it it prints the directories themselves. I tried this with `-not -path` which is another option but I couldn't get it to work as well, so I think this is better. You could add a comment explaining why `-print` is required but NAB

Reply:
added comment